diff --git a/src/Control/Reduce/Util/CliPredicate.hs b/src/Control/Reduce/Util/CliPredicate.hs
index 3c81070..37d4421 100644
--- a/src/Control/Reduce/Util/CliPredicate.hs
+++ b/src/Control/Reduce/Util/CliPredicate.hs
@@ -403,7 +403,7 @@ runCmd ::
 runCmd inputCreater workDir cmd a = do
   liftIO $ createDirectoryIfMissing True workDir
   withCurrentDirectory workDir $ do
-    (tp, _) <- timedPhase "setup" $ do
+    (tp, _) <- timedPhase ("setup " <> Builder.fromString workDir) $ do
       input <- inputCreater a
       liftIO $ setupWorkDir input cmd
     (tm, x) <- timedPhase "run" $ do
@@ -417,7 +417,7 @@ runCmd inputCreater workDir cmd a = do
             consumeWithHash
               (combineConsumers olog $ handlerLogger hout)
               (combineConsumers elog $ handlerLogger herr)
-              $ proc "./run.sh" []
+              $ proc "bash" ["./run.sh"]
     let m = fmap (formatResults tp tm) x
     logResults m
     return m
